frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePushBranchAction.java

```diff
@@ -84,7 +83,7 @@ public void update(AnActionEvent anActionEvent) {
   public void actionPerformed(AnActionEvent anActionEvent) {
 
     var project = getProject(anActionEvent);
-    var gitRepository = getSelectedVcsRepository(anActionEvent);
+    var gitRepository = project.getService(VcsRootComboBoxProvider.class).getSelectedVcsRepository();
```

Comment:
It could as well still be `var gitRepository = getSelectedVcsRepository(anActionEvent);` if `getSelectedVcsRepository` is moved to `IExpectsKeyProject` and implemented in terms of `getService`